### PR TITLE
DEPR: resample with PeriodIndex

### LINKED ISSUE #53481: DEPR: Resample with PeriodIndex?
Resampling with PeriodIndex is very very broken:

```
pi = pd.period_range(start="2000", periods=3, freq="M")
ser= pd.Series(["foo", "bar", "baz"], index=pi)
rs = ser.resample("M")  # should be a no-op!

>>> rs.nunique()
2000-01    foo
2000-02    bar
2000-03    baz
Freq: M, dtype: object
```

Some of this is because in PeriodIndexResampler._downsample (which many methods go through) we just return `self.asfreq()` in many cases which is very much wrong.  We have some xfailed tests (e.g. test_resample_nat_index_series) that cause this as well as some actively-incorrect tests (e.g. test_resample_empty_dtypes, test_resample_same_freq)

Is this worth fixing or should we just deprecate it?

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #53481 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.2.0.rst [hunk 1]
```diff
@@ -363,6 +363,7 @@ Other Deprecations
 - Deprecated :meth:`.DataFrameGroupBy.fillna` and :meth:`.SeriesGroupBy.fillna`; use :meth:`.DataFrameGroupBy.ffill`, :meth:`.DataFrameGroupBy.bfill` for forward and backward filling or :meth:`.DataFrame.fillna` to fill with a single value (or the Series equivalents) (:issue:`55718`)
 - Deprecated :meth:`Index.format`, use ``index.astype(str)`` or ``index.map(formatter)`` instead (:issue:`55413`)
 - Deprecated :meth:`Series.ravel`, the underlying array is already 1D, so ravel is not necessary (:issue:`52511`)
+- Deprecated :meth:`Series.resample` and :meth:`DataFrame.resample` with a :class:`PeriodIndex`, convert to :class:`DatetimeIndex` (with ``.to_timestamp()``) before resampling instead (:issue:`53481`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:366]: Might be good to mention that the `convention` will be removed too

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.2.0.rst]: good idea, updated+green

**Comment by jbrockmendel** [comment]
If there are no more comments, I plan to merge this in a week.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**